File: zip.js
// Printing frequency of each character just before its consecutive occurrences
const s = 'fffooewwwbbbb';
let out = '';

for (let i = 0; i < s.length; i++) {
  let count = 1;

  while (i + 1 < s.length && s.charAt(i) == s.charAt(i + 1)) {
    i++
    count++;
  }

  out += count + "" + s.charAt(i);
}

console.log(out);

File: zip.c
// Printing frequency of each character just before it's consecutive occurences
#include <stdio.h>

int main()
{
  char s[] = "fffooewwwbbbb";

  for (int i = 0; s[i] != 0; i++)
  {
    int count = 1;

    while (s[i] == s[i + 1]) {
      ++i;
      ++count;
    }

    printf("%d%c", count, s[i]);
  }

  return 0;
}